recipes/enable_chef_error_handler.rb

```diff
@@ -0,0 +1,13 @@
+
+cookbook_file "#{node['cluster']['scripts_dir']}/write_chef_error_handler.rb" do
+  source 'event_handler/write_chef_error_handler.rb'
+  owner 'root'
+  group 'root'
+  mode '0755'
+end
+
+chef_handler 'WriteChefError::WriteChefError' do
```

Comment:
Two comments here:

1. It seems we're using a chef resource that is deprecated: https://supermarket.chef.io/cookbooks/chef_handler - did you check the alternatives [here](https://docs.chef.io/handlers/#reporting) ?
2. If we're adding a dependency probably we should upgrade the Bersksfile accordingly and probably upload the dep in the `thirdy-party` cookbook folder.

Reply:
Thanks for the reminder. Correct me if I'm wrong (because I'm new to chef), but I feel that we are not using the deprecated chef_handler **cookbook**? 

(1) From your first link it says:

> "The chef_handler resource from **this cookbook** is **now shipping as part of Chef 14**. With **the inclusion of this resource into Chef itself** we are now deprecating this cookbook."

From the chef-client.log I see that we are using `chef_version=17.2.29`, so I'd assume we are not using the chef_handler **cookbook**? Instead, we are using resources from Chef itself? The alternative link you mentioned is indeed the webpage that I have been using as a reference. Another [webpage](https://docs.chef.io/resources/chef_handler/) about chef_handler resource also consistently mentions "New in Chef Infra Client 14.0". 

(2) If we were using the chef_handler **cookbook**, then where are the recipes from that cookbook (for example [these ones](https://github.com/chef-boneyard/chef_handler))? I don't find any in our codebase, or in the head node (under /etc/chef/cookbooks/), and also don't find clues when/where we download that cookbook.

Please let me know your thoughts and whether any additional actions are needed?

Reply:
Perfect, thanks for the detailed explanation. I'm aligned. We can keep it, thanks!